dot/core/messages.go

```diff
@@ -70,32 +74,32 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 		return false, err
 	}
 
-	allTxnsAreValid := true
+	allTxsAreValid := true
 	for _, tx := range txs {
-		txnIsValid := true
 		validity, err := s.validateTransaction(head, rt, tx)
 		if err != nil {
-			txnIsValid = false
-			allTxnsAreValid = false
-			switch err := err.(type) {
+			allTxsAreValid = false
+			switch err.(type) {
 			case runtime.InvalidTransaction:
 				s.net.ReportPeer(peerset.ReputationChange{
 					Value:  peerset.BadTransactionValue,
 					Reason: peerset.BadTransactionReason,
 				}, peerID)
+				continue
 			case runtime.UnknownTransaction:
+				continue
```

Comment:
Instead of this 
- you can put a single continue statement after switch statement finishes
or
- use `else if validity.Propagate {` and you won't need to continue.